src/Core/Manifest/Module.php

```diff
@@ -275,6 +293,79 @@ public function hasResource($path)
             ->getResource($path)
             ->exists();
     }
+
+    /**
+     * Determine what CI library the module is using.
+     * @internal
+     */
+    public function getCILibrary(): string
+    {
+        // We don't have any composer data at all
+        if (empty($this->composerData)) {
+            return self::CI_PHPUNIT_UNKNOWN;
+        }
+
+        // We don't have any dev dependencies
+        if (empty($this->composerData['require-dev']) || !is_array($this->composerData['require-dev'])) {
+            return self::CI_PHPUNIT_UNKNOWN;
+        }
+
+        // We are assuming a typical setup where the CI lib is defined in require-dev rather than require
+        $requireDev = $this->composerData['require-dev'];
+
+        // Try to pick which CI we are using based on phpunit constraint
+        $phpUnitConstraint = $this->requireDevConstraint(['sminnee/phpunit', 'phpunit/phpunit']);
+        if ($phpUnitConstraint) {
+            if ($this->satisfiesAtLeastOne(['5.7.0', '5.0.0', '5.x-dev', '5.7.x-dev'], $phpUnitConstraint)) {
```

Comment:
I'm giving a bunch of possible version here. If any one of them is matched, we assume that's the one the module is meant to use. If someone was super specific (e.g. requiring phpunit `5.7.32` or `~5.8.0`), none of the of those constraint would be matched and we'll return `CI_PHPUNIT_UNKNOWN`. Same thing for will apply for the PHPUnit 9 comparison.

This may or may not be a problem. My instinct is "not a problem" right now.

If we think it is a problem, we could try going a step further and using regex to extract the major release line of a very specific constraint.

Reply:
Just regex the contraint along the lines of `#^[~\^]?5#`?  Seems like this PR's a bit over-complicated due to the use of the `Semver::satisfiedBy()` over a simple regex

Reply:
My main fear here is that someone ends up writing a perfectly valid composer constraint that this thing can't understand. And there's so many ways to write composer constraint that there's no sensible regex that will capture all of them. That's why I used `Semver::satisfiedBy` out of the box.

The main edge case I'm concerned about is handling exact versions. I think we are better off trying `Semver::satisfiedBy` first and then falling back to some simplistic regex

Reply:
Yeah that makes sense to use regex only as a fallback.  I guess `"phpunit/phpunit": ">=5"` might actually be valid (haven't tried)

Reply:
Regex has been added as a supplement